src/Adapter/Form/ChoiceProvider/ZoneByIdChoiceProvider.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Form\ChoiceProvider;
+
+use PrestaShop\PrestaShop\Core\Form\ConfigurableFormChoiceProviderInterface;
+use Zone;
+
+/**
+ * This class will provide data from DB / ORM about Zone
+ */
+class ZoneByIdChoiceProvider implements ConfigurableFormChoiceProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getChoices(array $options): array
+    {
+        $active = isset($options['active']) && is_bool($options['active']) && $options['active'];
+
+        $activeFirst = isset($options['active_first']) && is_bool($options['active_first']) && $options['active_first'];
```

Comment:
If bool type is expected as option, an exception shouldn't get thrown in case of another type provided ?
```suggestion
        if (isset($options['active']) && !is_bool($options['active']) {
            new InvalidArgumentException(
                sprintf('"active" option type should be "bool", got "%s" type instead', gettype($options['active']));
        }
        $active = $options['active'] ?? false;

        if (isset($options['activeFirst']) && !is_bool($options['activeFirst']) {
            new InvalidArgumentException(
                sprintf('"activeFirst" option type should be "bool", got "%s" type instead', gettype($options['activeFirst']));
        }
        $activeFirst = $options['activeFirst'] ?? false;
```

Reply:
Not mandatory I'd say because the providers are mostly used in a certain context, in the core forms that respect the expected input options, but it's a good suggestion nonetheless (even if there are few chances for it to be used somewhere else or by anyone else)

Reply:
You could also use an `OptionsResolver` component to handle these input options

Reply:
`OptionResolver` is an excellent idea. Most helpful if it was included in a new `AbstractConfigurableChoiceProvider` class to extend, as a template method pattern, just like Symfony form types.